### ISSUE TITLE
Include a annotation to indicate contrast type in Contrast Map

### ISSUE THREAD

**Issue by nicholsn** [incf-nidash/nidm#243]
As discussed with @poldrack and @vsoch, currently [nidm:contrastMap](http://nidm.nidash.org/specs/nidm-results_020.html#section-nidm:ContrastMap) uses a string to denote the contrast with the attribute `nidm:contrastName`; however, what we want to use is a URI that defines the specific contrast - ideally from an ontology with additional knowledge about the contrast.

The Cognitive Atlas provides a knowledgebase of `concepts` (e.g., [response inhibition](http://www.cognitiveatlas.org/concept/id/trm_4a3fd79d0af66)), that are measured by `tasks` (e.g., [stop signal task](http://www.cognitiveatlas.org/task/id/tsk_4a57abb949e1a)), which have specific `contrasts` (e.g., successful stop trial - go trial). There are also links to datasets that use these contrasts (e.g., https://openfmri.org/dataset/ds000007)

Each of the concepts, tasks, contrasts, etc. have a URI, thus by "semantically" annotating a `nidm:ContrastMap` with using an attribute-contrast tag (something like `cogat:contrast` `cogat:MyContrastURI`), we could link out to access the specific task and what mental process is being measured, etc.

Thoughts?

**Comment by poldrack** [comment]
I have an email in to the cog atlas developers to see about contrast uris, I thought they had been implemented but I'm not certain
Cheers
Rp

**Comment by nicholsn** [comment]
@poldrack, that would be a nice addition if its not in the system already.

**Comment by vsoch** [comment]
@chrisfilo connected me to @r03ert0 to talk about a data structure that would be a good start to getting these different things all connected.  I have an idea, and today I'm trying out a first go at making a connection between cogatlas URIs, maps in neurovault, and the data standard used by brainspell.  Not a lot of details in this note - but I'll send out an update in the next few days!

**Comment by poldrack** [comment]
It turns out that we do have contrast Uris but they are currently not exposed and also not restful. They are going to look into exposing them for us
Cheers Rp

**Comment by nicholsn** [comment]
great, thanks!

**Comment by vsoch** [comment]
Is there a flat file / temporary version laying around that I can use for the time being? I can write my scripts without it, but probably would want the "one for realsie" in a day or so!

**Comment by vsoch** [comment]
And please disregard the above message while it is a Sunday :L)

**Comment by poldrack** [comment]
Yes, I can dump the db to a file. Will try to do that ASAP but I'm traveling now

**Comment by vsoch** [comment]
Awesome!! Yeah no worries, I won't need it for a bit! Safe travels.

**Comment by cmaumet** [comment]
Hi! Having an additional attribute in the `nidm:contrastMap` to link to a Cognitive Atlas "contrast" would be great! Though I would not discard the `nidm:contrastName` attribute (in a form of a string) as this is probably the only piece of information we can currently automatically extract from an analysis (SPM, FSL, AFNI)?. 

We could also add "cognitive atlas" attributes in the `nidm:DesignMatrix` entity.

**Comment by dbkeator** [comment]
Hi,
I'm thinking it might be nice to have a more general attribute or create the attribute such that one could link a URI from "any" ontology that defines the contrast.  So, by "any" I of course mean ones that are thought to be mature and relevant to our work.  I do think this is a nice addition though.
